code/game/gamemodes/gang/gang.dm

```diff
@@ -92,6 +113,83 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
+
+	return 1
+
+/datum/game_mode/gang/proc/gangpocalypse()
+	set waitfor = FALSE
+	..(FALSE, FALSE)
+	var/list/bosses = list()
+	for(var/datum/gang/G in gangs)
+		for(var/datum/mind/boss_mind in G.bosses)
+			bosses += boss_mind
+			//var/msg = "<span class='danger'>You are a local leader for the [G.name] gang, with security gone this station is ripe for the taking!"
+			forge_gang_objectives(boss_mind)
+			greet_gang(boss_mind)
+			G.add_gang_hud(boss_mind)
+			equip_gang(boss_mind.current,G)
+			modePlayer += boss_mind
+	sleep(50)
+	priority_announce("Excessive costs associated with lawsuits from employees injured by Security and Synthetic crew have compelled us to re-evaluate the personnel budget for new stations. Accordingly, this station will be expected to operate without Security or Synthetic assistance.", "Nanotrasen Board of Directors")
+	sleep(70)
+	priority_announce("Unfortunately we have also received reports of multiple criminal enterprises established in your sector. To assist in repelling this threat, we have implanted all crew with a device that will assist and incentivize the removal of all contraband and criminals. Enjoy your shift ", "Nanotrasen Board of Directors")
+	sleep(30)
+	explosion(target_armory, 10, 16, 28, 30, TRUE, TRUE)
+	explosion(target_equipment, 10, 16, 28, 30, TRUE, TRUE)
+
+	for(var/mob/living/M in GLOB.player_list)
+		if(!is_gangster(M))
+			vigilize(M)
+
+/proc/gangtest()
+	var/turf/target_armory
+	var/turf/target_equipment
+	for(var/area/ai_monitored/security/armory/A in GLOB.sortedAreas)
+		if(A.z == ZLEVEL_STATION)
+			target_armory = pick(get_area_turfs(A))
+			if(target_armory)
+				break
+	//also need to find 1 sec closet to hit the equipment room
+	for(var/area/security/main/A in GLOB.sortedAreas)
+		if(A.z == ZLEVEL_STATION)
+			var/obj/structure/closet/secure_closet/security/C = locate() in A
+			if(C)
+				target_equipment = get_turf(C)
+				break
+
+	if(!target_armory || !target_equipment)
+		message_admins("No armory/equipment room detected, unable to start gang!")
+		return FALSE
+	sleep(20)
+	priority_announce("Excessive costs associated with lawsuits from employees injured by Security and Synthetic crew have compelled us to re-evaluate the personnel budget for new stations. Accordingly, this station will be expected to operate without Security or Synthetic assistance.", "Nanotrasen Board of Directors")
+	sleep(60)
+	priority_announce("Unfortunately we have also received reports of several notorious gangs established in your sector. To assist in repelling this threat, we have taken the liberty of implanting you all with a device that will activate in the event of a gang incursion. Enjoy your shift.", "Nanotrasen Board of Directors")
+	sleep(40)
+	explosion(target_armory, 10, 16, 28, 30, TRUE, TRUE)
+	explosion(target_equipment, 10, 16, 28, 30, TRUE, TRUE)
+	for(var/mob/living/M in GLOB.player_list)
+		vigilize(M)
+
+/proc/vigilize(mob/living/M)
```

Comment:
I will, I made it global for debugging since I can't get Gang War to start during testing because of other checks that would be more dangerous to remove for testing purposes.